src/Compilers/CSharp/Portable/Symbols/Source/LocalFunctionOrSourceMemberMethodSymbol.cs

```diff
@@ -46,10 +46,22 @@ internal sealed override TypeWithAnnotations IteratorElementTypeWithAnnotations
 
         /// <summary>
         /// Whether the method has the <see langword="unsafe"/> keyword in its signature.
-        /// Do not confuse with <see cref="IsCallerUnsafe"/>.
+        /// Do not confuse with <see cref="CallerUnsafeMode"/>.
         /// </summary>
         internal abstract bool IsUnsafe { get; }
 
-        internal sealed override bool IsCallerUnsafe => ContainingModule.UseUpdatedMemorySafetyRules && IsUnsafe;
+        internal sealed override CallerUnsafeMode CallerUnsafeMode
```

Comment:
Consider pushing this override up into the base (`SourceMethodSymbol`) as well as `IsUnsafe`. Then we just need `override IsUnsafe => false` in `LambdaSymbol` (which we may adjust later if we allow the `unsafe` keyword).